shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java

```diff
@@ -312,9 +308,11 @@ public List<SimpleSubjectDTO> findAllSubjectsOfStudyId(final Long studyId) {
 	@Override
 	public List<SimpleSubjectDTO> findAllSubjectsOfStudyAndPreclinical(final Long studyId, boolean preclinical) {
 		List<SimpleSubjectDTO> simpleSubjectDTOList = new ArrayList<>();
-		List<SubjectStudy> opt = subjectStudyRepository.findByStudy(studyRepository.findById(studyId).orElse(null));
-		if (opt != null) {
-			for (SubjectStudy rel : opt) {
+		List<SubjectStudy> studySubjects = subjectStudyRepository.findByStudyId(studyId);
+		Study studyWithTags = studyRepository.findStudyWithTagsById(studyId);
+		studySubjects.stream().forEach(ss -> ss.setStudy(studyWithTags));
+		if (studySubjects != null) {
```

Comment:
Maybe return empty List after line 311 if studySubjects is null, to limit cyclomatic complexity ?